Add polo 1.0

agonzalezro opened the pull request:
Polo is a static site generator written in Go and compatible with Jekyll
& Pelican markdown files.

DomT4 commented on Library/Formula/polo.rb:
```diff
@@ -0,0 +1,22 @@
+    ENV["GOPATH"] = buildpath
+    ENV["GOHOME"] = buildpath
+
+    system "go", "get", "github.com/constabulary/gb/..."
```
Please use the `go_resource` method instead of `go get`. Example [here](https://github.com/Homebrew/homebrew/blob/master/Library/Formula/consul-template.rb#L131-L144).

agonzalezro replied:
Hi @DomT4, I tried with `go_resource` in the past, but the difference between the example I saw and this script is that I will need a binary generated by `go get` to be able to build `polo`.

If `go_resource` is going to create this binary for me, where is it placed? I couldn't find it.

DomT4 replied:
It should be installed locally to that directory. You can see lots of use of `godep` in this way in various formulae and [here](https://github.com/Homebrew/homebrew/blob/master/Library/Formula/wellington.rb#L60-L63).

agonzalezro commented:
I addressed the comments to use `go_resource` instead. Thanks for the comments.

bfontaine commented on Library/Formula/polo.rb:
```diff
@@ -0,0 +1,31 @@
+  end
+
+  test do
+    system "#{bin}/polo"
```
We could write a test that generates a dummy blog. Upstream provides an example source directory here: https://github.com/agonzalezro/polo/tree/master/example.

agonzalezro commented:
@bfontaine I added the text using the `example` folder, but TBH I don't think that this kind of test belongs here. I think that it should be a integration test in `polo` itself. But if it's how you usually do it I have no problems at all.

@dunn I've addressed the comments.

Thanks!

bfontaine commented on Library/Formula/polo.rb:
```diff
@@ -0,0 +1,33 @@
+  test do
+    system "wget", "https://github.com/agonzalezro/polo/archive/v1.0.tar.gz"
+    system "tar", "zxf", "v1.0.tar.gz"
+    system "cd", "polo-1.0/example", ";", "#{bin}/polo", ".", "/tmp/polo-test"
```
You don’t need the whole archive:
```rb
(testpath/"config.json").write <<-EOS.undent
  {
    "author": "Foo Bar",
    "title": "test blog"
  }
EOS

system "#{bin}/polo", ".", "myblog"
assert File.directory? "myblog"
```

agonzalezro replied:
Awesome! I was not really happy downloading the full archive (pretty slow test) but you asked me to use the `example` and I blind follow it.

Thanks for the snippet. I will test it and update the PR.

bfontaine replied:
Yeah I was looking for a way to download the example alone but we can’t without re-downloading the archive; Tried with the `config.json` file only and it worked :smile:

bfontaine commented on Library/Formula/polo.rb:
```diff
@@ -0,0 +1,39 @@
+  desc "Static blog generator compatible with Jekyll & Pelican content"
+  homepage "https://github.com/agonzalezro/polo"
+  url "https://github.com/agonzalezro/polo/archive/v1.0.tar.gz"
+  sha256 "efe7e798212b05b76ab63f32603bc15eedd9eadec3bfbfc95e36c64ba13f00dc"
```
Seems like it’s the wrong checksum:
```
==> Downloading https://github.com/agonzalezro/polo/archive/v1.0.tar.gz
==> Downloading from https://codeload.github.com/agonzalezro/polo/tar.gz/v1.0
Warning: Formula reports different SHA256: efe7e798212b05b76ab63f32603bc15eedd9eadec3bfbfc95e36c64ba13f00dc
Downloaded to: /Library/Caches/Homebrew/polo-1.0.tar.gz
SHA1: 96644f22ffd3f9237c3ea0a0b5dfe87f81a1c630
SHA256: 39ba03b17f725faf319c97a3e9e5852e4abb8c420e3cd4216b7c1a0ba7dcb7ff
```

agonzalezro commented:
Sorry about the checksum. I obviously did something wrong.

agonzalezro commented:
I was wondering if you need something else from my side to get this PR ready.

bfontaine commented:
@BrewTestBot test this please.

DomT4 commented on Library/Formula/polo.rb:
```diff
@@ -0,0 +1,39 @@
+    Language::Go.stage_deps resources, buildpath/"src"
+
+    cd "src/github.com/constabulary/gb" do
+      system "go", "get", "./..."
```
Can you explain this step? `go get` usually fetches things remotely. Not the case here, I presume?

agonzalezro replied:
Hi @DomT4 

`polo` is using [gb](getgb.io) as dependency manager. As you can see on their page, they require a `... get`:

    go get github.com/constabulary/gb/...

And this is the only way I found here to do so.

DomT4 replied:
So there's no remote fetching happening here at all?

agonzalezro replied:
Probably yes. It will download the `gb` dependencies into the `$GOPATH`, in
this case `buildpath`.

I tried to find another example about how to do it on `homebrew` codebase
but I couldn't find anything.

On Wed, Nov 18, 2015 at 8:04 PM, Dominyk Tiller <notifications@github.com>
wrote:

> In Library/Formula/polo.rb
> <https://github.com/Homebrew/homebrew/pull/44796#discussion_r45243759>:
>
> > +  url "https://github.com/agonzalezro/polo/archive/v1.0.tar.gz"
> > +  sha256 "39ba03b17f725faf319c97a3e9e5852e4abb8c420e3cd4216b7c1a0ba7dcb7ff"
> > +
> > +  depends_on "go" => :build
> > +
> > +  go_resource "github.com/constabulary/gb" do
> > +    url "https://github.com/constabulary/gb.git",
> > +        :revision => "23f405f2981056a97abb52b5b7e528a6cddef1bb"
> > +  end
> > +
> > +  def install
> > +    ENV["GOPATH"] = buildpath
> > +    Language::Go.stage_deps resources, buildpath/"src"
> > +
> > +    cd "src/github.com/constabulary/gb" do
> > +      system "go", "get", "./..."
>
> So there's no remote fetching happening here at all?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Homebrew/homebrew/pull/44796/files#r45243759>.
>



-- 
@agonzalezro <http://twitter.com/agonzalezro> or my blog
<http://agonzalezro.github.io>
Please, don't send me files with extensions: .doc, .docx, .xls, .xlsx, .ppt
and/or .pptx

DomT4 replied:
We may need to try and use `go_resource`s here :confused:. We don't do remote fetching during compile, mostly because it makes caching things and building offline later impossible, and means if the build breaks once you're having to re-download things you downloaded a minute ago afresh.

Not sure. Maybe other maintainers have opinions here.

dunn replied:
Are the dependencies fetched by `gb` the ones specified in the vendor/manifest? https://github.com/agonzalezro/polo/blob/master/vendor/manifest

If so let's translate those into `go_resources`, thanks!

bfontaine replied:
+1 for `go_resource`s as well.

bfontaine commented:
Ping?

agonzalezro commented:
Sorry, I am working on a new version of polo and I will get rid of gb.

I will update this PR when I finish it.

Close it now if you want me to reopen when I finish or let it open.

On Mon, Dec 14, 2015 at 3:11 PM, Baptiste Fontaine <notifications@github.com
> wrote:

> Ping?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Homebrew/homebrew/pull/44796#issuecomment-164447092>.
>



-- 
@agonzalezro <http://twitter.com/agonzalezro> or my blog
<http://agonzalezro.github.io>
Please, don't send me files with extensions: .doc, .docx, .xls, .xlsx, .ppt
and/or .pptx

bfontaine commented:
It’s fine to leave it open if you’re working on it :wink:

bfontaine commented:
Gentle ping :smile:

agonzalezro commented:
Hi @bfontaine, thanks for the ping. Trust me, this is still on my TODO list (ongoing, I did some stuff), but I understand that it's taking too much time so feel free to close it and reopen it later? If you don't mind keeping it open, let's go for that solution.

mikemcquaid commented:
Closing for now; feel free to recreate on `homebrew-core` at a later point and relink to this one. Thanks!
